crit/mempages.go

```diff
@@ -193,3 +194,69 @@ func (mr *MemoryReader) GetShmemSize() (int64, error) {
 
 	return size, nil
 }
+
+// PatternMatch represents a match when searching for a pattern in memory.
+type PatternMatch struct {
+	Vaddr   uint64
+	Length  int
+	Context int
+	Pattern string
+}
+
+// SearchPattern searches for a pattern in the process memory pages.
+// If the pattern is found, it returns slices of strings containing the matched patterns
+// along with the specified context (number of bytes before and after each match).
+func (mr *MemoryReader) SearchPattern(pattern string, context int) ([]PatternMatch, error) {
+	if context < 0 {
+		return nil, errors.New("context size cannot be negative")
+	}
+
+	regexPattern, err := regexp.Compile(pattern)
+	if err != nil {
+		return nil, err
+	}
+
+	var results []PatternMatch
+
+	for _, entry := range mr.pagemapEntries {
+		startAddr := entry.GetVaddr()
+		endAddr := startAddr + uint64(entry.GetNrPages())*uint64(mr.pageSize)
+
+		// Read the entire pages content
+		buffer, err := mr.GetMemPages(startAddr, endAddr)
+		if err != nil {
+			return nil, err
+		}
+
+		// Replace null bytes with space
+		pagesContent := bytes.ReplaceAll(buffer.Bytes(), []byte{0x0}, []byte{0x20})
```

Comment:
This looks interesting, may I know why we're doing this? I can't seem to think of a reason to eliminate null bytes from the memory page contents.

Reply:
I have found that null bytes can cause regex to either match the wrong pattern or exceed the intended matching pattern, resulting in unpredictable behavior. Replacing them with space ensures we match the intended pattern.

Reply:
You should add that information as a comment. I often here a comment should describe why you it and not what you do. So instead `Replace null bytes with space` it would be good to have the explanation why you replace the null bytes.